spec/services/discovery_report_spec.rb

```diff
@@ -55,5 +55,59 @@
         druid: 'druid:kk203bw3276'
       )
     end
+
+    context "folders are empty" do
+      it "adds empty_object error" do
+        expect(report.process_dobj(dobj)).to match a_hash_including(errors: a_hash_including(empty_object: true))
+      end
+    end
+
+    context "folders are not empty" do
+      let(:obj_file) { instance_double(PreAssembly::ObjectFile, path: "random/path", filesize: 324, mimetype: "")}
+
+      before do
+        allow(dobj).to receive(:object_files).and_return([obj_file, obj_file])
+        allow(report).to receive(:using_smpl_manifest?).and_return(false)
+        allow(report).to receive(:registration_check).and_return({}) # pretend everything is in Dor
+      end
+
+      it "does not add empty_object error" do
+        expect(report.process_dobj(dobj)).not_to include(a_hash_including(empty_object: true))
+      end
+    end
+  end
+
+  context "integration test" do
+    let(:bc_params) do
+      {
+        project_name: "SmokeTest",
+        content_structure: 0,
+        bundle_dir: 'spec/test_data/images_jp2_tif',
+        staging_style_symlink: false,
+        content_metadata_creation: 0,
+        user: build(:user, sunet_id: 'jdoe@stanford.edu')
+      }
+    end
+    let(:bundle_context) {BundleContext.new(bc_params) }
+    let(:bundle) { PreAssembly::Bundle.new(bundle_context) }
+    let(:dobj) { report.bundle.objects_to_process.first }
+
+    before do
+      bundle.manifest_rows.each { |row| row.merge!("object" => row["folder"]) }
```

Comment:
this line isn't be needed any longer - please remove it  (you may need to rebase first)